### PR TITLE
Pick libmonosgen-2.0.so from cmake install directory instead of .libs

### PR DESCRIPTION
**Pull request by akoeplinger**
This aligns Linux with what we already do for all the other platforms.

### CODE REVIEW

File: src/mono/mono.proj [hunk 1]
```diff
@@ -417,7 +417,8 @@
       <_MonoRuntimeFilePath Condition="'$(TargetstvOS)' == 'true'">$(MonoObjDir)out\lib\libmonosgen-2.0.dylib</_MonoRuntimeFilePath>
       <_MonoRuntimeFilePath Condition="'$(TargetsAndroid)' == 'true'">$(MonoObjDir)out\lib\libmonosgen-2.0.so</_MonoRuntimeFilePath>
       <_MonoRuntimeFilePath Condition="'$(TargetsBrowser)' == 'true'">$(MonoObjDir)out\lib\libmonosgen-2.0.a</_MonoRuntimeFilePath>
-      <_MonoRuntimeFilePath Condition="'$(_MonoRuntimeFilePath)' == ''">$(MonoObjDir)mono\mini\.libs\libmonosgen-2.0.so</_MonoRuntimeFilePath>
+      <_MonoRuntimeFilePath Condition="'$(_MonoRuntimeFilePath)' == '' and '$(TargetsLinux)' == 'true' and '$(TargetArchitecture)' == 'x64'">$(MonoObjDir)out\lib64\libmonosgen-2.0.so</_MonoRuntimeFilePath>
```
Comment (CoffeeFlux) [inline src/mono/mono.proj:420]: Shouldn't we be trying to move more of this to CMake rather than less?

Comment (akoeplinger) [reply src/mono/mono.proj:420]: I didn't find an easy way to overwrite the `lib` value that cmake uses for the install directory.

Comment (vargaz) [reply src/mono/mono.proj:420]: Why is this needed, are those files different ?

File: src/mono/mono/mini/CMakeLists.txt [hunk 2]
```diff
@@ -304,7 +304,6 @@ endif()
 if(NOT DISABLE_SHARED_LIBS)
   add_library(monosgen SHARED $<TARGET_OBJECTS:monosgen-objects>)
   set_target_properties(monosgen PROPERTIES OUTPUT_NAME monosgen-2.0)
-  set_target_properties(monosgen PROPERTIES LIBRARY_OUTPUT_DIRECTORY .libs)
   target_link_libraries(monosgen ${OS_LIBS} ${ICONV_LIB} ${LLVM_LIBS} ${ICU_LIBS})
```
Comment (vargaz) [inline src/mono/mono/mini/CMakeLists.txt:307]: Not sure why this was here in the first place.

Comment (akoeplinger) [reply src/mono/mono/mini/CMakeLists.txt:307]: it replicated what the automake-based build did, outputting libraries into .libs

File: src/mono/mono.proj (hunk 1 above)
Comment (akoeplinger) [reply src/mono/mono.proj:420]: not really, but they could be if you add custom install rules. the main thing is it makes us consistent accross platforms.

File: src/mono/mono.proj [hunk 3]
```diff
@@ -417,7 +417,8 @@
       <_MonoRuntimeFilePath Condition="'$(TargetstvOS)' == 'true'">$(MonoObjDir)out\lib\libmonosgen-2.0.dylib</_MonoRuntimeFilePath>
       <_MonoRuntimeFilePath Condition="'$(TargetsAndroid)' == 'true'">$(MonoObjDir)out\lib\libmonosgen-2.0.so</_MonoRuntimeFilePath>
       <_MonoRuntimeFilePath Condition="'$(TargetsBrowser)' == 'true'">$(MonoObjDir)out\lib\libmonosgen-2.0.a</_MonoRuntimeFilePath>
-      <_MonoRuntimeFilePath Condition="'$(_MonoRuntimeFilePath)' == ''">$(MonoObjDir)mono\mini\.libs\libmonosgen-2.0.so</_MonoRuntimeFilePath>
+      <_MonoRuntimeFilePath Condition="'$(_MonoRuntimeFilePath)' == '' and Exists('$(MonoObjDir)out\lib64\libmonosgen-2.0.so')">$(MonoObjDir)out\lib64\libmonosgen-2.0.so</_MonoRuntimeFilePath>
+      <_MonoRuntimeFilePath Condition="'$(_MonoRuntimeFilePath)' == ''">$(MonoObjDir)out\lib\libmonosgen-2.0.so</_MonoRuntimeFilePath>
```
Comment (vargaz) [inline src/mono/mono.proj:421]: Doesn't cmake have a property to control the name of this directory ?

Comment (akoeplinger) [reply src/mono/mono.proj:421]: yeah that works too, will change.

File: src/mono/mono.proj [hunk 4]
```diff
@@ -417,7 +418,7 @@
       <_MonoRuntimeFilePath Condition="'$(TargetstvOS)' == 'true'">$(MonoObjDir)out\lib\libmonosgen-2.0.dylib</_MonoRuntimeFilePath>
       <_MonoRuntimeFilePath Condition="'$(TargetsAndroid)' == 'true'">$(MonoObjDir)out\lib\libmonosgen-2.0.so</_MonoRuntimeFilePath>
       <_MonoRuntimeFilePath Condition="'$(TargetsBrowser)' == 'true'">$(MonoObjDir)out\lib\libmonosgen-2.0.a</_MonoRuntimeFilePath>
```
Comment (vargaz) [inline src/mono/mono.proj:420]: The android line is not needed any more.

Comment (akoeplinger) [reply src/mono/mono.proj:419]: Fixed.

**Review by vargaz — APPROVED** [review]

**Pull request merged**